torch/csrc/autograd/python_variable_indexing.cpp

```diff
@@ -304,63 +242,86 @@ static THPObjectPtr wrapTuple(PyObject* index) {
   return res;
 }
 
+// NOTE: Here is the dispatch structure for `THPVariable_getitem`:
+//
+// 1. Python 1-D getter calls C++ `at::indexing::get_item` after
+// converting Python index to C++ TensorIndex, except `t[...]` and `t[None]`
+// in which we perform direct operations on `t` instead.
+//
+// 2. Python N-D getter calls C++ `at::indexing::handleDimInMultiDimIndexing`
+// for each dim, after converting Python index to C++ TensorIndex. If advanced
+// indexing is needed, it calls C++ `at::indexing::dispatch_index`.
 PyObject* THPVariable_getitem(PyObject* self, PyObject* index) {
   HANDLE_TH_ERRORS
   auto& self_ = reinterpret_cast<THPVariable*>(self)->cdata;
   OptionalDeviceGuard device_guard(device_of(self_));
 
-  // handle simple types: integers, slices, ellipsis
+  // handle simple types: none, ellipsis
   if (index == Py_None) {
-    return wrap(self_.unsqueeze(0));
+    return THPVariable_Wrap(self_.unsqueeze(0));
```

Comment:
For my understanding, could this now be `handle_simple_type(at::indexing::TensorIndex(Ellipses), /*release_gil*/ false)`?

Reply:
I found that using the `handle_simple_type` lambda actually adds ~30ns overhead to `v[None]` or `v[...]`, so I changed them to directly calling `THPVariable_Wrap(at::indexing::get_item(self_, {at::indexing::TensorIndex(at::indexing::None)}))`